avr-specs/sync-from-upstream.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/python3
+
+import subprocess
+import json
+import os
+
+ENCODING = 'utf-8'
+
+
+def main():
+    process = subprocess.Popen(['rustc', '--print', 'target-spec-json', '-Z', 'unstable-options', '--target', 'avr-unknown-gnu-atmega328'],
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE)
+
+    ref_spec_string, stderr = process.communicate()
```

Comment:
Instead of using `Popen`, you should probably use [`subprocess.run()`](https://docs.python.org/3/library/subprocess.html#subprocess.run).  Also, leave stderr as it is so errors from the compiler are shown on the console.  And lastly, using `check=True` here makes sure that errors are caught and not ignored.

```suggestion
    ref_spec_string = subprocess.run(
        ['rustc', '--print', 'target-spec-json', '-Z', 'unstable-options', '--target', 'avr-unknown-gnu-atmega328'],
        check=True,
        stdout=subprocess.PIPE,
    ).stdout
```